### PR TITLE
Attempt to drop items safely when acid blocking chests

### PR DESCRIPTION
**Pull request by Diet-Cola**

### CODE REVIEW

**Comment by Diet-Cola** [comment]
#62

File: src/main/java/vg/civcraft/mc/citadel/command/Acid.java [hunk 1]
```diff
@@ -105,4 +110,17 @@ public boolean execute(CommandSender sender, String[] args) {
 	public List<String> tabComplete(CommandSender sender, String[] args) {
 		return new ArrayList<>();
 	}
+
+	public boolean isChestBeingAcidBlocked (Block block) {
```
Comment (Maxopoly) [inline src/main/java/vg/civcraft/mc/citadel/command/Acid.java]: A method name like this makes it sound like this just checks something, but it actually actively drops stuff. Definitely should be changed.

File: src/main/java/vg/civcraft/mc/citadel/command/Acid.java [hunk 2]
```diff
@@ -105,4 +110,17 @@ public boolean execute(CommandSender sender, String[] args) {
...
+		if (block.getType() != Material.CHEST || block.getType() != Material.TRAPPED_CHEST) {
```
Comment (Maxopoly) [inline src/main/java/vg/civcraft/mc/citadel/command/Acid.java]: Should use some other check to see if its an instance of container or something like that, not just those two types of chests

File: src/main/java/vg/civcraft/mc/citadel/command/Acid.java [hunk 3]
```diff
@@ -105,4 +110,17 @@ public boolean execute(CommandSender sender, String[] args) {
...
+			return false;
+		}
+		Chest chest = (Chest) block.getBlockData();
+		ItemStack[] items =  chest.getBlockInventory().getStorageContents();
+		chest.getBlockInventory().clear();
```
Comment (Maxopoly) [inline src/main/java/vg/civcraft/mc/citadel/command/Acid.java]: Did you test all kinds of edge conditions involving double chests with this?

Comment (Diet-Cola) [reply src/main/java/vg/civcraft/mc/citadel/command/Acid.java]: Did some basic testing, single chests alone dropped contents ok
double chest when acid blocking left side dropped nothing but chest 
acid blocking a double chest at once also dropped correct inventory

File: src/main/java/vg/civcraft/mc/citadel/command/Acid.java (hunk 1 above)
Comment (Diet-Cola) [reply src/main/java/vg/civcraft/mc/citadel/command/Acid.java]: canDropChestContents more suited?

File: src/main/java/vg/civcraft/mc/citadel/command/Acid.java (hunk 3 above)
Comment (Diet-Cola) [reply src/main/java/vg/civcraft/mc/citadel/command/Acid.java]: seems to work okay from what ive tested as basic uses of it, is there anything complex you want me to test about it?

Comment (Maxopoly) [reply src/main/java/vg/civcraft/mc/citadel/command/Acid.java]: It should drop half when you acid one half. Each half of a double chest inventory belongs to one of the chests

File: src/main/java/vg/civcraft/mc/citadel/command/Acid.java (hunk 1 above)
Comment (Maxopoly) [reply src/main/java/vg/civcraft/mc/citadel/command/Acid.java]: No, that also sounds like a getter. Just make all of it a breakBlock() method and include the handling for non-inventories

File: src/main/java/vg/civcraft/mc/citadel/command/Acid.java (hunk 3 above)
Comment (Diet-Cola) [reply src/main/java/vg/civcraft/mc/citadel/command/Acid.java]: Didn't fill the chest enough on first try, this works as is

**Comment by Maxopoly** [comment]
Did you ensure that this does not dupe ShulkerBoxes? Those are also Container.

**Comment by Diet-Cola** [comment]
Seems to work just fine

**Comment by Maxopoly** [comment]
Ok, very nice then
